hybrid/ddc_compartment_switching/ddc_compartment_switching_nok.c

```diff
@@ -0,0 +1,41 @@
+/***
+ * This example shows an intermediate level of compartment creation.
+ * It allocates a block of memory (5000 bytes) but only the first
+ * 2000 are accessible by `compartment_simple_fun`.
+ *
+ * This program rises an exception because we try to access an
+ * `int32_t` which occurs to be out of the compartment's bounds.
+ ***/
+
+#include "../../include/common.h"
+#include "../include/utils.h"
+#include <assert.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#if !defined(__CHERI_CAPABILITY_WIDTH__) || defined(__CHERI_PURE_CAPABILITY__)
+#error "This example only works on CHERI hybrid mode"
+#endif
+
+int compartment_simple_fun();
+extern int switch_compartment(void *stack, size_t size);
+
+int main()
+{
+	int32_t *simple_block = malloc(5000);
+	size_t compartment_size = 2000;
+	simple_block[700] = 8;
+	switch_compartment(simple_block, compartment_size);
+	return 0;
+}
+
+int compartment_simple_fun()
+{
+	int32_t *__capability ddc_cap = cheri_ddc_get();
+	// This function can access only 2000 bytes, i.e. `compartment_size`
+	assert(cheri_tag_get(ddc_cap) && cheri_length_get(ddc_cap) == 2000);
+	// This function cannot go over its compartment bounds
+	assert(ddc_cap[700]);
```

Comment:
This `assert` doesn't look right: either the derefence fails because it's out of bounds, or we're testing that `assert(8 != 0)` which the compiler might even be able to statically prove is true.

Reply:
Maybe a comment and single assertion could be better/more clear. In cb8307f.